### PR TITLE
Trend line on details page

### PR DESCRIPTION
**Pull request by jamesuejio**
- Trend line on details page
- Reverse event order
- add start, end, calendar_ids to query
- limit pie graph and average hours to 2 decimal points

### CODE REVIEW

File: cal/cal/models.py [hunk 1]
```diff
@@ -550,7 +550,7 @@ def __str__(self):
     def hours(self, calendar_ids=None, start=None, end=None):
         events = self.get_events(calendar_ids=calendar_ids, start=start, end=end)
 
-        return EventCollection(lambda: events).total_time() / 3600
+        return round(float(EventCollection(lambda: events).total_time()) / 3600, 2)
```
Comment (MaxFangX) [inline cal/cal/models.py:556]: Why do we do this?

File: cal/cal/models.py [hunk 2]
```diff
@@ -620,7 +620,7 @@ def save(self, *args, **kwargs):
...
```
Comment (MaxFangX) [inline cal/cal/models.py:626]: Could you perhaps add a comment here explaining why this is needed? For me at least it was not readily apparent

File: cal/cal/static/js/services.js [hunk 3]
```diff
@@ -220,24 +220,69 @@ analyticsApp.service('QueryService', function() {
...
+      color: '#FDB515',
+      strokeWidth: 3,
+    });
+
     return [ctrlDetails, maxYValue];
   };
 });
+
+// returns slope, intercept and r-square of the line
+function leastSquares(xSeries, ySeries) {
```
Comment (MaxFangX) [inline cal/cal/static/js/services.js]: global functions should go in `calendaranalytics.js`, but I think you should just attach this to `QueryService` instead.

**Review by MaxFangX — APPROVED** [review]
After you've implemented the tiny changes, please merge! This is great stuff - I'm excited. I love that you added a leastSquares function LOL

And thanks for installing the linter too ;)

File: cal/cal/models.py (hunk 1 above)
Comment (jamesuejio) [reply cal/cal/models.py:556]: This is necessary to round to two decimal places. Before the pie chart was rounded down, but this becomes sort of confusing when you have pie charts per week and all your hours are off by 30 mins or something.

File: cal/cal/models.py (hunk 2 above)
Comment (jamesuejio) [reply cal/cal/models.py:626]: ok got it. I did it in the javascript file too so I'll add a comment as well.

File: cal/cal/static/js/services.js (hunk 3 above)
Comment (jamesuejio) [reply cal/cal/static/js/services.js]: Do you mean saying this.leastSquares = ...? Or just putting it directly in the this.populateData function.

Comment (cindytung) [reply cal/cal/static/js/services.js]: You can have another function in QueryService and call it by doing QueryService.leastSquares. You'll also have to make sure whatever controller is using this function takes in QueryService as a controller in its component

Comment (jamesuejio) [reply cal/cal/static/js/services.js]: I just copy and pasted it and stuck it in. Max said not to bind to controller just make it a helper function.

**Pull request merged**